Sources/TuistGraph/Models/Destination.swift

```diff
@@ -31,15 +31,15 @@ public enum Destination: String, Codable, Equatable, CaseIterable {
 
     public var platform: Platform {
         switch self {
-        case .iPad, .iPhone, .macCatalyst, .macWithiPadDesign, .appleVisionWithiPadDesign:
+        case .iPad, .iPhone, .macCatalyst, .macWithiPadDesign:
             return .iOS
         case .mac:
             return .macOS
         case .appleTv:
             return .tvOS
         case .appleWatch:
             return .watchOS
-        case .appleVision:
+        case .appleVision, .appleVisionWithiPadDesign:
```

Comment:
Changing the association with `platform` will cause issues that diverge with what Xcode derives from these settings.  If a target has a single destination of `appleVisionWithiPadDesign` the `SDKROOT` is `iphoneos` but this would cause tuist to use `xros` instead.   Additionally, `SUPPORTED_PLATFORMS` would resolve to `xros xrsimulator`.

There are a few other values that are derived from this value, but making a this change would be a meaningful change to how `tuist` generates targets and I dont think this is the right approach to allowing `XROS_DEPLOYMENT_TARGET` to be set for `appleVisionWithiPadDesign ` targets.

I would expect that adjusting the [TargetLinter](https://github.com/tuist/tuist/blob/362ef5e3c35fda201960ecadc5edde9ef0ca5734/Sources/TuistGenerator/Linter/TargetLinter.swift#L208) to account for this need is a more narrow change to accomplish the goal.

Reply:
I ran some tests and it worked absolutely fine, but generally I agree - I wanted to do the more narrow change in TargetLinter but because it checks platform.supports, it will mean I manually have to check against the target's destinations which also seems a bit gnarly.

I'll add another commit with that option.